agroal-pool/src/main/java/io/agroal/pool/ConnectionPool.java

```diff
@@ -488,7 +516,7 @@ public void returnConnectionHandler(ConnectionHandler handler) throws SQLExcepti
 
         if ( handler.setState( CHECKED_OUT, CHECKED_IN ) ) {
             // here the handler is already up for grabs
-            synchronizer.releaseConditional();
+            handlerTransferQueue.tryTransfer( handler );
```

Comment:
same here?

Reply:
I see your point, and I've been thinking about it, but the thing is that the calls are not symmetric. an acquire is not required to call `poll()`  and if every connection return would call `offer()` there would be a huge leak.

`LinkedTransferQueue` is not bound either (that could be an option to limit the number of connections on the queue) and calling `size()` is an expensive operation.